src/modules/fancyzones/lib/ZoneWindow.cpp

```diff
@@ -10,6 +10,63 @@
 
 #include <ShellScalingApi.h>
 
+namespace {
+  enum class TTmpFileType {
+    EActiveZoneSetFile = 0,
+    EAppliedZoneSetFile,
+    ECustomZoneSetsFile
+  };
+
+  std::wstring GenerateTmpFileName(TTmpFileType type)
+  {
+    static std::wstring activeZoneSetTmpFileName;
+    static std::wstring appliedZoneSetTmpFileName;
+    static std::wstring customZoneSetsTmpFileName;
+
+    if (activeZoneSetTmpFileName.empty()) {
```

Comment:
let's use std::call_once instead of `empty()` checks to ensure thread-safety just in case.